src/components/RepoBranchesDrawer/RepoBranchesDrawer.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+
+import { Drawer } from 'antd';
+
+import GitHubStore from '@/store/GitHubStore';
+import { RepoItem } from '@/store/GitHubStore/types';
+
+type DrawerOnCloseEvent = React.KeyboardEvent<HTMLDivElement> | React.MouseEvent<HTMLDivElement | HTMLButtonElement>;
+
+export type RepoBranchesDrawerProps = {
+    selectedRepo: RepoItem | null;
+    visible: boolean;
+    onClose: (e: DrawerOnCloseEvent) => void;
+};
+
+const RepoBranchesDrawer: React.FC<RepoBranchesDrawerProps> = ({ selectedRepo, visible, onClose }) => {
+    const [branches, setBranches] = useState<string[]>([]);
+
+    useEffect(() => {
+        if (selectedRepo === null) {
+            setBranches([]);
+            return;
+        }
+
+        let mounted = true;
```

Comment:
Читал [статьи](https://www.digitalocean.com/community/tutorials/how-to-handle-async-data-loading-lazy-loading-and-code-splitting-with-react) про best practices, наткнулся на такой прием: 
"When you unmount, the component useEffect updates the variable. The asynchronous function will still resolve, but it won’t make any changes to unmounted components. This will prevent memory leaks."
Но если так не надо делать, то я уберу